### PR TITLE
Added Aakash's pipeline and implemented minor changes

### PR DESCRIPTION
**Pull request by sayakpaul**
Reference: https://github.com/sayakpaul/Handwriting-Recognizer-in-Keras/issues/1

### CODE REVIEW

File: handwriting_recognition.ipynb [hunk 1]
```diff
@@ -0,0 +1,1027 @@
+{
```
Comment (AakashKumarNain) [inline handwriting_recognition.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.    def prepare_image_paths(samples):
</pre><p>Can we rename this to something like <code>get_image_path_and_labels(..)</code>?  Any name that we use in the code should be self-explanatory as much as possible</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/Handwriting-Recognizer-in-Keras/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (AakashKumarNain) [inline handwriting_recognition.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #27.    train_paths, train_labels = prepare_image_paths(train_samples)
</pre><p>Similary, rename <code>train_paths</code>  to <code>train_img_paths</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/Handwriting-Recognizer-in-Keras/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (AakashKumarNain) [inline handwriting_recognition.ipynb:1]: <ol><li>Aspect ratio is preserved</li><li>Content of the image isn't distorted</li></ol><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sayakpaul/Handwriting-Recognizer-in-Keras/pull/2/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

**Comment by AakashKumarNain** [comment]
@sayakpaul I guess the model won't take much time to train. Let us train it for at least 50-60 epochs

**Comment by AakashKumarNain** [comment]
This what I got after 34 epochs. Pretty good results IMO

**Comment by sayakpaul** [comment]
> @sayakpaul I guess the model won't take much time to train. Let us train it for at least 50-60 epochs

Each epoch takes about 48 seconds to run. So, 50 epochs would take (48 * 50) / 60 = 40 minutes. For generating the additional files, this might be an overkill.

**Comment by AakashKumarNain** [comment]
> > @sayakpaul I guess the model won't take much time to train. Let us train it for at least 50-60 epochs
> 
> Each epoch takes about 48 seconds to run. So, 50 epochs would take (48 * 50) / 60 = 40 minutes. For generating the additional files, this might be an overkill.

Cool. Let's go with 30 only then

**Comment by sayakpaul** [comment]
Sounds good. Here's what I am thinking about the next steps:

* I incorporate 30 epochs of training.
* Generate the `.py` file and then we review it amongst ourselves.
* Once things look good we create the PR. 

Sounds good?

**Comment by AakashKumarNain** [comment]
Sounds good to me. Also, I think you left one more change that is required. For cleaning the labels, just define a function `clean_labels(..)` and call it for training labels, validation labels, and test labels. This would also help reduce the LOC

**Comment by AakashKumarNain** [comment]
Side note: I am not sure if all the ops here are supported by TFLite but we can give it a try to convert it into a model than runs on phone. It would be awesome if we can do that. We can even showcase it in the code example if we are able to accomplish it

**Comment by sayakpaul** [comment]
> Sounds good to me. Also, I think you left one more change that is required. For cleaning the labels, just define a function clean_labels(..) and call it for training labels, validation labels, and test labels. This would also help reduce the LOC

Done. However, I kept the training label cleaner as it is because it is printing and storing important information which is being used later on. I did not want to have multiple variables returning from the utility and wanted to keep it clean.

> Side note: I am not sure if all the ops here are supported by TFLite but we can give it a try to convert it into a model than runs on phone. It would be awesome if we can do that. We can even showcase it in the code example if we are able to accomplish it

Yes. It's possible to convert to TFLite. Refer to the following notebooks:

* https://github.com/tulasiram58827/ocr_tflite/blob/main/colabs/captcha_ocr_tflite.ipynb
* https://github.com/tulasiram58827/ocr_tflite/blob/main/colabs/captcha_ocr_inference.ipynb

**Comment by AakashKumarNain** [comment]
This LGTM! If you want to make more changes, do let me know else we can merge this
